Sources/Sentry/SentrySerialization.m

```diff
@@ -162,6 +162,23 @@ + (SentryEnvelope *_Nullable) envelopeWithData:(NSData *)data {
     return envelope;
 }
 
++ (NSData *_Nullable)dataWithSession:(SentrySession *)session
+                              options:(NSJSONWritingOptions)opt
+                                error:(NSError *_Nullable *_Nullable)error {
+    return [self dataWithJSONObject:[session serialize] options:opt error:error];
+}
+
++ (SentrySession *_Nullable)sessionWithData:(NSData *)sessionData {
+    NSError *error = nil;
+    NSDictionary *sessionDictionary = [NSJSONSerialization JSONObjectWithData:sessionData options:0 error:&error];
+    if (nil != error) {
+        [SentryLog logWithMessage:[NSString stringWithFormat:@"Failed to deserialize session data %@", error] andLevel:kSentryLogLevelError];
+        return nil;
+    }
+    SentrySession *session = [[SentrySession alloc] initWithJSONObject:sessionDictionary];
+    return session;
+}
+
```

Comment:
What about putting this functionality in an extra class? I don't really like this class here as it does too many things. I wouldn't add more stuff to it.

Reply:
This was recently added so we don't have multiple calls to NSJSON with the guard to check for valid JSON etc.
So far it serializes Sentry types so I wouldn't say it does multiple things, it's a single responsability.
How would you suggest we split?